core/arch/riscv64/emit_utils.c

```diff
@@ -540,8 +574,28 @@ fragment_ibt_prefix_size(uint flags)
 void
 insert_fragment_prefix(dcontext_t *dcontext, fragment_t *f)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
+    /* Always use prefix on RISCV64 as there is no load to PC. */
+    byte *write_start = vmcode_get_writable_addr(f->start_pc);
+    byte *pc = write_start;
+    ASSERT(f->prefix_size == 0);
```

Comment:
nit: move the ASSERT to the top of the function.

Reply:
Done.